### ISSUE TITLE
`no-unnecessary-boolean-comparison`: does not take into account possible non-boolean values

### ISSUE THREAD

**Issue by andreww2012** [sindresorhus/eslint-plugin-unicorn#3385]
The both `if` statement expressions are reported which should not:

```ts
const foo = ({bar = false}: {bar?: boolean | 'baz' | ''}) => {
  if (bar === false) {
   // ^^^^^^^^^^^^^
  }
  if (bar === true) {
   // ^^^^^^^^^^^^^
  }
};
```
